packages/vc-verification/src/resolver/Ipfs-credential-resolver.ts

```diff
@@ -193,34 +193,32 @@ export class IpfsCredentialResolver implements CredentialResolver {
   ): Promise<RoleEIP191JWT[]> {
     const didDocument = await this.getDIDDocument(did, didDocumentCache);
     const services: IServiceEndpoint[] = didDocument.service || [];
-    return (
-      await Promise.all(
-        services.map(async ({ serviceEndpoint }) => {
-          if (!isCID(serviceEndpoint)) {
-            return {};
-          }
+    const resolved: Array<RoleEIP191JWT> = [];
+    for (const { serviceEndpoint } of services) {
```

Comment:
Performance issue. Consider Promise.allSettled().

Reply:
this is for purpose. When there are too many claims ( I tested with 30), then all these requests are sent at once and IPFS returs error something like `too many p2p requests`

Reply:
How about introducing `p-limit` or similar approach?